File: Array/1816. Truncate Sentence/TruncateSentence.java
class Solution {
    public String truncateSentence(String s, int k) {
        StringBuilder result = new StringBuilder();
        for (char ch : s.toCharArray()) {
            if (ch == ' ') {
                k--;
            }
            if (k == 0) {
                break;
            }
            result.append(ch);
        }
        return result.toString();
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        String str = "Hello how are you Contestant";
        int k = 4;
        String result = sol.truncateSentence(str, k);
        System.out.println("Truncated sentence: " + result);
    }
}

File: Array/1816. Truncate Sentence/TruncateSentence.go
package main

import (
	"fmt"
)

func truncateSentence(s string, k int) string {
	result := ""
	for _, ch := range s {
		if ch == ' ' {
			k--
		}
		if k == 0 {
			break
		}
		result += string(ch)
	}
	return result
}

func main() {
	s := "Hello how are you Contestant"
	k := 4
	result := truncateSentence(s, k)
	fmt.Println("Truncated sentence:", result)
}
